scanpy/preprocessing/_normalization.py

```diff
@@ -35,7 +35,7 @@ def _normalize_data(X, counts, after=None, copy: bool = False):
     elif isinstance(counts, np.ndarray):
         np.divide(X, counts[:, None], out=X)
     else:
-        X = np.divide(X, counts[:, None])  # dask does not support kwarg "out"
+        X = X / counts[:, None]
```

Comment:
This division op causes a `COO` matrix to be spit out.  We could map blocks back to CSR or write a custom division function at a lower level.  However, there's no guarantee on the blocks of X and those of counts really matching up...

Reply:
I'm not sure if this is a blocker or not...also just mapping the blocks of `X` to `csr_matrix` or changing `_meta` back also doesn't seem to work amazingly

Reply:
One option would be to `map_blocks` but use `block_id` and `block_info` to do array access in the `func` passed in.  But I'm not sure how this plays with the parallel model since `counts` would not really be mapped....

Reply:
I'm finding this very difficult.  If this coo business is a bug, I think we can just run with "it's a bug, hopefully it's fixed soon"

Reply:
> However, there's no guarantee on the blocks of X and those of counts really matching up...

Could we just enforce this ourselves?

Reply:
@ivirshup I guess this is something I considered but it more likely would mean rechunking `X` than `counts` which could be expensive?  I guess not though so maybe it's not so bad?

Another option would be to `compute` `counts` and just reference the bug?  I'll try again with rechunking, didn't actually try it before discounting it...

Reply:
I would assume in most cases `counts` should have a chunk structure like `X` since it was computed from `X`, no?

Reply:
> X (since it might be broken up on the var axis)

Shouldn't chunking along the `var` axis not mater, since chunking should match along `obs`? I believe `map_blocks` should allow broadcasting when a dimension is missing.

Reply:
I agree that it what it says in theory, but in practice this was not what happened....I will need to try again